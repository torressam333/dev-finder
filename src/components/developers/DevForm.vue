<template>
  <form @submit.prevent="formSubmitted">
    <div class="form-control">
      <label for="firstname">First Name</label>
      <input type="text" id="firstname" v-model.trim="firstName.value" />
    </div>
    <div class="form-control">
      <label for="lastname">Last Name</label>
      <input type="text" id="lastname" v-model.trim="lastName.value" />
    </div>
    <div class="form-control">
      <label for="description">Description</label>
      <textarea
        type="text"
        id="description"
        rows="5"
        v-model.trim="description.value"
      ></textarea>
    </div>
    <div class="form-control">
      <label for="rate">Hourly Rate</label>
      <input type="number" id="rate" v-model.number="hourlyRate.value" />
    </div>
    <div class="form-control">
      <h3>Areas of Expertise</h3>
      <div>
        <input
          type="checkbox"
          id="frontend"
          value="frontend"
          v-model="areas.value"
        />
        <label for="frontend">Frontend Developer</label>
      </div>
      <div>
        <input
          type="checkbox"
          id="backend"
          value="backend"
          v-model="areas.value"
        />
        <label for="backend">Backend Developer</label>
      </div>
      <div>
        <input
          type="checkbox"
          id="machineLearning"
          value="machineLearning"
          v-model="areas.value"
        />
        <label for="machineLearning">Machine Learning Engineer</label>
      </div>
    </div>
    <base-button>Register</base-button>
  </form>
</template>

<script>
import BaseButton from '../base-components/BaseButton.vue';
export default {
  name: 'dev-form',
  emits: ['save-data'],
  components: { BaseButton },
  data() {
    return {
      areas: {
        value: [],
        isValid: true,
      },
      description: {
        value: '',
        isValid: true,
      },
      firstName: {
        value: '',
        isValid: true,
      },
      formIsValid: true,
      hourlyRate: {
        value: null,
        isValid: true,
      },
      lastName: {
        value: '',
        isValid: true,
      },
    };
  },
  methods: {
    formSubmitted() {
      this.validateForm();

      if (!this.formIsValid) {
        return;
      }

      const formData = {
        areas: this.areas.value,
        desc: this.description.value,
        first: this.firstName.value,
        last: this.lastName.value,
        rate: this.hourlyRate.value,
      };

      this.$emit('save-data', formData);
    },
    validateForm() {
      this.formIsValid = true;

      //Check each fields value, if any one fails then the
      //entire form is invalid
      if (!this.firstName.value) {
        this.firstName.isValid = false;
        this.formIsValid = false;
      }

      if (!this.lastName.value) {
        this.lastName.isValid = false;
        this.formIsValid = false;
      }

      if (!this.description.value) {
        this.description.isValid = false;
        this.formIsValid = false;
      }

      if (!this.hourlyRate.value || this.hourlyRate.val < 0) {
        this.hourlyRate.isValid = false;
        this.formIsValid = false;
      }

      if (!this.areas.value.length) {
        this.areas.isValid = false;
        this.formIsValid = false;
      }
    },
  },
};
</script>

<style scoped>
.form-control {
  margin: 0.5rem 0;
}

label {
  display: block;
  font-weight: bold;
  margin-bottom: 0.5rem;
}

input[type='checkbox'] + label {
  display: inline;
  font-weight: normal;
  margin: 0 0 0 0.5rem;
}

input,
textarea {
  border-radius: 5px;
  border: 1px solid #ccc;
  display: block;
  font: inherit;
  width: 100%;
}

input:focus,
textarea:focus {
  background-color: #e2eff5;
  border-color: #00658d;
  outline: none;
}

input[type='checkbox'] {
  border: none;
  display: inline;
  width: auto;
}

input[type='checkbox']:focus {
  outline: #00658d solid 1px;
}

h3 {
  font-size: 1rem;
  margin: 0.5rem 0;
}

.invalid label {
  color: rgb(223, 35, 35);
}

.invalid input,
.invalid textarea {
  border: 1px solid rgb(223, 35, 35);
}
</style>
