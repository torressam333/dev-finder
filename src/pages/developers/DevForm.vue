<template>
  <form action="">
    <div class="form-control">
      <label for="firstname">Firstname</label>
      <input type="text" id="firstname" />
    </div>
    <div class="form-control">
      <label for="lastname">LastName</label>
      <input type="text" id="lastname" />
    </div>
    <div class="form-control">
      <label for="description">Description</label>
      <textarea type="text" id="description" rows="5"></textarea>
    </div>
    <div class="form-control">
      <label for="rate">Hourly Rate</label>
      <input type="number" id="rate" />
    </div>
    <div class="form-control">
      <h3>Areas of Expertise</h3>
      <div>
        <input type="checkbox" id="frontend" value="frontend" />
        <label for="frontend">Frontend Developer</label>
      </div>
      <div>
        <input type="checkbox" id="backend" value="backend" />
        <label for="backend">Backend Developer</label>
      </div>
      <div>
        <input type="checkbox" id="machineLearning" value="machineLearning" />
        <label for="machineLearning">Machine Learning Engineer</label>
      </div>
    </div>
    <base-button>Register</base-button>
  </form>
</template>
