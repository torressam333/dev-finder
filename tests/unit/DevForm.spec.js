import { shallowMount } from '@vue/test-utils';
import DevForm from '../../src/components/developers/DevForm.vue';

let wrapper;

beforeEach(() => {
  wrapper = shallowMount(DevForm);
});

test('should render html properly', () => {
  expect(wrapper).toMatchSnapshot();
});
