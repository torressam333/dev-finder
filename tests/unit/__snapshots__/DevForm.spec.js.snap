// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render html properly 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <form
        action=""
      >
        <div
          class="form-control"
        >
          <label
            for="firstname"
          >
            Firstname
          </label>
          <input
            id="firstname"
            type="text"
          />
        </div>
        <div
          class="form-control"
        >
          <label
            for="lastname"
          >
            LastName
          </label>
          <input
            id="lastname"
            type="text"
          />
        </div>
        <div
          class="form-control"
        >
          <label
            for="description"
          >
            Description
          </label>
          <textarea
            id="description"
            rows="5"
            type="text"
          />
        </div>
        <div
          class="form-control"
        >
          <label
            for="rate"
          >
            Hourly Rate
          </label>
          <input
            id="rate"
            type="number"
          />
        </div>
        <div
          class="form-control"
        >
          <h3>
            Areas of Expertise
          </h3>
          <div>
            <input
              id="frontend"
              type="checkbox"
              value="frontend"
            />
            <label
              for="frontend"
            >
              Frontend Developer
            </label>
          </div>
          <div>
            <input
              id="backend"
              type="checkbox"
              value="backend"
            />
            <label
              for="backend"
            >
              Backend Developer
            </label>
          </div>
          <div>
            <input
              id="machineLearning"
              type="checkbox"
              value="machineLearning"
            />
            <label
              for="machineLearning"
            >
              Machine Learning Engineer
            </label>
          </div>
        </div>
        <base-button>
          Register
        </base-button>
      </form>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
        },
        "transition-group": Object {
          "name": "transition-group",
        },
      },
      "config": Object {
        "compilerOptions": Object {},
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "emitsCache": WeakMap {},
      "mixins": Array [
        Object {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": Object {},
      "reload": [Function],
    },
    "_instance": Object {
      "a": null,
      "accessCache": Object {},
      "appContext": Object {
        "app": [Circular],
        "components": Object {
          "transition": Object {
            "name": "transition",
          },
          "transition-group": Object {
            "name": "transition-group",
          },
        },
        "config": Object {
          "compilerOptions": Object {},
          "errorHandler": undefined,
          "globalProperties": Object {},
          "isNativeTag": [Function],
          "optionMergeStrategies": Object {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": Object {},
        "emitsCache": WeakMap {},
        "mixins": Array [
          Object {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": Object {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": Object {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": Object {},
      "da": null,
      "data": Object {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": Array [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": Object {},
      "propsDefaults": Object {},
      "propsOptions": Array [
        Object {},
        Array [],
      ],
      "provides": Object {},
      "proxy": Object {},
      "refs": Object {
        "VTU_COMPONENT": Object {
          "hasOwnProperty": [Function],
        },
      },
      "render": [Function],
      "renderCache": Array [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": Array [],
        "effects": Array [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
      },
      "setupContext": null,
      "setupState": Object {},
      "slots": Object {},
      "sp": null,
      "subTree": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": Object {
          "a": null,
          "accessCache": Object {
            "@@__IMMUTABLE_ITERABLE__@@": 0,
            "@@__IMMUTABLE_RECORD__@@": 0,
            "_isMockFunction": 0,
            "constructor": 0,
            "hasOwnProperty": 4,
            "isVueInstance": 0,
            "toJSON": 0,
            Symbol(Symbol.toStringTag): 0,
          },
          "appContext": Object {
            "app": [Circular],
            "components": Object {
              "transition": Object {
                "name": "transition",
              },
              "transition-group": Object {
                "name": "transition-group",
              },
            },
            "config": Object {
              "compilerOptions": Object {},
              "errorHandler": undefined,
              "globalProperties": Object {},
              "isNativeTag": [Function],
              "optionMergeStrategies": Object {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": Object {},
            "emitsCache": WeakMap {},
            "mixins": Array [
              Object {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": Object {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": Object {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": Object {},
          "ctx": Object {
            "hasOwnProperty": [Function],
          },
          "da": null,
          "data": Object {},
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": null,
          "exposed": null,
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": Object {},
          "propsDefaults": Object {},
          "propsOptions": Array [
            Object {},
            Array [],
          ],
          "provides": Object {},
          "proxy": Object {
            "hasOwnProperty": [Function],
          },
          "refs": Object {},
          "render": [Function],
          "renderCache": Array [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": Array [],
            "effects": Array [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": Array [],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
          },
          "setupContext": null,
          "setupState": Object {},
          "slots": Object {},
          "sp": null,
          "subTree": Object {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": <div
                  class="form-control"
                >
                  <h3>
                    Areas of Expertise
                  </h3>
                  <div>
                    <input
                      id="frontend"
                      type="checkbox"
                      value="frontend"
                    />
                    <label
                      for="frontend"
                    >
                      Frontend Developer
                    </label>
                  </div>
                  <div>
                    <input
                      id="backend"
                      type="checkbox"
                      value="backend"
                    />
                    <label
                      for="backend"
                    >
                      Backend Developer
                    </label>
                  </div>
                  <div>
                    <input
                      id="machineLearning"
                      type="checkbox"
                      value="machineLearning"
                    />
                    <label
                      for="machineLearning"
                    >
                      Machine Learning Engineer
                    </label>
                  </div>
                </div>,
                "appContext": null,
                "children": <div class="form-control"><label for="firstname">Firstname</label><input type="text" id="firstname"></div>
<div class="form-control"><label for="lastname">LastName</label><input type="text" id="lastname"></div>
<div class="form-control"><label for="description">Description</label><textarea type="text" id="description" rows="5"></textarea></div>
<div class="form-control"><label for="rate">Hourly Rate</label><input type="number" id="rate"></div>
<div class="form-control">
  <h3>Areas of Expertise</h3>
  <div><input type="checkbox" id="frontend" value="frontend"><label for="frontend">Frontend Developer</label></div>
  <div><input type="checkbox" id="backend" value="backend"><label for="backend">Backend Developer</label></div>
  <div><input type="checkbox" id="machineLearning" value="machineLearning"><label for="machineLearning">Machine Learning Engineer</label></div>
</div>,
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <div
                  class="form-control"
                >
                  <label
                    for="firstname"
                  >
                    Firstname
                  </label>
                  <input
                    id="firstname"
                    type="text"
                  />
                </div>,
                "key": null,
                "patchFlag": 0,
                "props": null,
                "ref": null,
                "scopeId": null,
                "shapeFlag": 8,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 5,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": Symbol(Static),
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": Array [
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "Register",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": Register,
                    "key": null,
                    "patchFlag": 0,
                    "props": null,
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 8,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": Symbol(Text),
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <base-button>
                  Register
                </base-button>,
                "key": null,
                "patchFlag": 0,
                "props": null,
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "base-button",
              },
            ],
            "component": null,
            "dirs": null,
            "dynamicChildren": Array [],
            "dynamicProps": null,
            "el": <form
              action=""
            >
              <div
                class="form-control"
              >
                <label
                  for="firstname"
                >
                  Firstname
                </label>
                <input
                  id="firstname"
                  type="text"
                />
              </div>
              <div
                class="form-control"
              >
                <label
                  for="lastname"
                >
                  LastName
                </label>
                <input
                  id="lastname"
                  type="text"
                />
              </div>
              <div
                class="form-control"
              >
                <label
                  for="description"
                >
                  Description
                </label>
                <textarea
                  id="description"
                  rows="5"
                  type="text"
                />
              </div>
              <div
                class="form-control"
              >
                <label
                  for="rate"
                >
                  Hourly Rate
                </label>
                <input
                  id="rate"
                  type="number"
                />
              </div>
              <div
                class="form-control"
              >
                <h3>
                  Areas of Expertise
                </h3>
                <div>
                  <input
                    id="frontend"
                    type="checkbox"
                    value="frontend"
                  />
                  <label
                    for="frontend"
                  >
                    Frontend Developer
                  </label>
                </div>
                <div>
                  <input
                    id="backend"
                    type="checkbox"
                    value="backend"
                  />
                  <label
                    for="backend"
                  >
                    Backend Developer
                  </label>
                </div>
                <div>
                  <input
                    id="machineLearning"
                    type="checkbox"
                    value="machineLearning"
                  />
                  <label
                    for="machineLearning"
                  >
                    Machine Learning Engineer
                  </label>
                </div>
              </div>
              <base-button>
                Register
              </base-button>
            </form>,
            "key": null,
            "patchFlag": 0,
            "props": Object {
              "action": "",
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "form",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": Object {
            "components": Object {},
            "render": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <form
          action=""
        >
          <div
            class="form-control"
          >
            <label
              for="firstname"
            >
              Firstname
            </label>
            <input
              id="firstname"
              type="text"
            />
          </div>
          <div
            class="form-control"
          >
            <label
              for="lastname"
            >
              LastName
            </label>
            <input
              id="lastname"
              type="text"
            />
          </div>
          <div
            class="form-control"
          >
            <label
              for="description"
            >
              Description
            </label>
            <textarea
              id="description"
              rows="5"
              type="text"
            />
          </div>
          <div
            class="form-control"
          >
            <label
              for="rate"
            >
              Hourly Rate
            </label>
            <input
              id="rate"
              type="number"
            />
          </div>
          <div
            class="form-control"
          >
            <h3>
              Areas of Expertise
            </h3>
            <div>
              <input
                id="frontend"
                type="checkbox"
                value="frontend"
              />
              <label
                for="frontend"
              >
                Frontend Developer
              </label>
            </div>
            <div>
              <input
                id="backend"
                type="checkbox"
                value="backend"
              />
              <label
                for="backend"
              >
                Backend Developer
              </label>
            </div>
            <div>
              <input
                id="machineLearning"
                type="checkbox"
                value="machineLearning"
              />
              <label
                for="machineLearning"
              >
                Machine Learning Engineer
              </label>
            </div>
          </div>
          <base-button>
            Register
          </base-button>
        </form>,
        "key": null,
        "patchFlag": 0,
        "props": Object {
          "ref": "VTU_COMPONENT",
        },
        "ref": Object {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "components": Object {},
          "render": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": Object {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": Object {
          "app": [Circular],
          "components": Object {
            "transition": Object {
              "name": "transition",
            },
            "transition-group": Object {
              "name": "transition-group",
            },
          },
          "config": Object {
            "compilerOptions": Object {},
            "errorHandler": undefined,
            "globalProperties": Object {},
            "isNativeTag": [Function],
            "optionMergeStrategies": Object {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": Object {},
          "emitsCache": WeakMap {},
          "mixins": Array [
            Object {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": Object {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <form
          action=""
        >
          <div
            class="form-control"
          >
            <label
              for="firstname"
            >
              Firstname
            </label>
            <input
              id="firstname"
              type="text"
            />
          </div>
          <div
            class="form-control"
          >
            <label
              for="lastname"
            >
              LastName
            </label>
            <input
              id="lastname"
              type="text"
            />
          </div>
          <div
            class="form-control"
          >
            <label
              for="description"
            >
              Description
            </label>
            <textarea
              id="description"
              rows="5"
              type="text"
            />
          </div>
          <div
            class="form-control"
          >
            <label
              for="rate"
            >
              Hourly Rate
            </label>
            <input
              id="rate"
              type="number"
            />
          </div>
          <div
            class="form-control"
          >
            <h3>
              Areas of Expertise
            </h3>
            <div>
              <input
                id="frontend"
                type="checkbox"
                value="frontend"
              />
              <label
                for="frontend"
              >
                Frontend Developer
              </label>
            </div>
            <div>
              <input
                id="backend"
                type="checkbox"
                value="backend"
              />
              <label
                for="backend"
              >
                Backend Developer
              </label>
            </div>
            <div>
              <input
                id="machineLearning"
                type="checkbox"
                value="machineLearning"
              />
              <label
                for="machineLearning"
              >
                Machine Learning Engineer
              </label>
            </div>
          </div>
          <base-button>
            Register
          </base-button>
        </form>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "compilerOptions": Object {},
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.30",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "hasOwnProperty": [Function],
  },
  "isDisabled": [Function],
  "rootVM": Object {},
  "wrapperElement": <form
    action=""
  >
    <div
      class="form-control"
    >
      <label
        for="firstname"
      >
        Firstname
      </label>
      <input
        id="firstname"
        type="text"
      />
    </div>
    <div
      class="form-control"
    >
      <label
        for="lastname"
      >
        LastName
      </label>
      <input
        id="lastname"
        type="text"
      />
    </div>
    <div
      class="form-control"
    >
      <label
        for="description"
      >
        Description
      </label>
      <textarea
        id="description"
        rows="5"
        type="text"
      />
    </div>
    <div
      class="form-control"
    >
      <label
        for="rate"
      >
        Hourly Rate
      </label>
      <input
        id="rate"
        type="number"
      />
    </div>
    <div
      class="form-control"
    >
      <h3>
        Areas of Expertise
      </h3>
      <div>
        <input
          id="frontend"
          type="checkbox"
          value="frontend"
        />
        <label
          for="frontend"
        >
          Frontend Developer
        </label>
      </div>
      <div>
        <input
          id="backend"
          type="checkbox"
          value="backend"
        />
        <label
          for="backend"
        >
          Backend Developer
        </label>
      </div>
      <div>
        <input
          id="machineLearning"
          type="checkbox"
          value="machineLearning"
        />
        <label
          for="machineLearning"
        >
          Machine Learning Engineer
        </label>
      </div>
    </div>
    <base-button>
      Register
    </base-button>
  </form>,
}
`;
